api/etcd/get_test.go

```diff
@@ -0,0 +1,69 @@
+package etcd
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/heetch/regula"
+	"github.com/heetch/regula/api"
+	"github.com/heetch/regula/rule"
+	"github.com/stretchr/testify/require"
+)
+
+func TestGet(t *testing.T) {
+	t.Parallel()
+
+	s, cleanup := newEtcdRulesetService(t)
+	defer cleanup()
+
+	path := "p/a/t/h"
+	sig := &regula.Signature{ReturnType: "bool", Params: make(map[string]string)}
+
+	t.Run("OK", func(t *testing.T) {
+		rules1 := []*rule.Rule{rule.New(rule.True(), rule.BoolValue(true))}
+		createBoolRuleset(t, s, path, rules1...)
+
+		ruleset1, err := s.Get(context.Background(), path, "")
+		require.NoError(t, err)
+		require.Equal(t, path, ruleset1.Path)
+		require.Len(t, ruleset1.Versions, 1)
+		require.NotEmpty(t, rules1, ruleset1.Version)
+		require.Equal(t, rules1, ruleset1.Rules)
+		require.Equal(t, sig, ruleset1.Signature)
+
+		// we are waiting 1 second here to avoid creating the new version in the same second as the previous one
```

Comment:
:tulip: given the constraint, we should perhaps guard against creating more than one per second if we wish to maintain version ordering this way (KSUID is a good solution, I'm not questioning that).

Reply:
Indeed, that's a very good point. I was thinking that in terms of UI, if two people update the same ruleset at the same moment, there's no real value in knowing who pressed the save button first. But if you use Regula's API to programmatically create rules sequentially you will lose the expected ordering.
I've created an issue for that: https://github.com/heetch/regula/issues/120